core/src/main/java/hudson/model/Queue.java

```diff
@@ -713,7 +723,16 @@ public boolean add(Task p, int quietPeriod, Action... actions) {
     }
 
     /**
-     * Cancels the item in the queue. If the item is scheduled more than once, cancels the first occurrence.
+     * Cancels the item in the queue. If the item is scheduled more than once,
+     * cancels the <b>first</b> occurrence.
+     * <p>
+     * Be aware, that the same task can be multiple times in the Queue, e.g. if
+     * parameters are set on the project. In that case, it is Russian Roulette,
+     * whether or not you cancel the correct task.
```

Comment:
So does it make sense to deprecate this overload, or are there cases (at least from searching usages within this repo) where it is impractical to use the other overloads?

There are several new methods being introduced, but zero examples of them being used successfully, either here or in some downstream PR using an incremental deployment. Any proposed new API must have at least one matching proposed call (outside of test code) which proves that the API is useful. If none can be found, then the method should be deleted. And if no proposes usage can be found outside of this repository (in a plugin), then it should be

```java
@Restricted(NoExternalUse.class)
```

unless and until one is found.